src/JSInterop/Microsoft.JSInterop/test/Infrastructure/DotNetDispatcherTest.cs

```diff
@@ -314,6 +314,110 @@ public async Task EndInvoke_WithNullError()
         Assert.Empty(ex.Message);
     }
 
+    [Fact]
+    public void EndInvokeJS_DoesNotThrowJSONExceptionIfTaskCancelled()
+    {
+        // Arrange
+        var jsRuntime = new TestJSRuntime();
+        var testDTO = new TestDTO { StringVal = "Hello", IntVal = 4 };
+        var cts = new CancellationTokenSource();
+        var argsJson = JsonSerializer.Serialize(new object[] { jsRuntime.LastInvocationAsyncHandle, true, testDTO }, jsRuntime.JsonSerializerOptions);
```

Comment:
Do we need an other test where the json is malformed but the cts is canceled prior to it being read?